paasta_tools/setup_marathon_job.py

```diff
@@ -417,9 +418,16 @@ def log_deploy_error(errormsg, level='event'):
     )
 
     if new_app_running:
+        hosts_tasks_running_on = [task.host for task in new_app.tasks]
+        draining_hosts = get_draining_hosts()
+        num_at_risk_tasks = 0
+        for host in hosts_tasks_running_on:
+            if host in draining_hosts:
+                num_at_risk_tasks += 1
+        log.debug("%s has %d tasks running on at-risk hosts." % (new_app.id, num_at_risk_tasks))
         protected_draining_tasks = set()
-        if new_app.instances < config['instances']:
-            client.scale_app(app_id=new_app.id, instances=config['instances'], force=True)
+        if new_app.instances < config['instances'] + num_at_risk_tasks:
+            client.scale_app(app_id=new_app.id, instances=config['instances'] + num_at_risk_tasks, force=True)
         elif new_app.instances > config['instances']:
```

Comment:
Can do. I still need to add logic to drain the at-risk tasks (rather than letting it choose a potentially healthy/not at-risk task to drain).